torch/csrc/jit/runtime/register_special_ops.cpp

```diff
@@ -293,7 +294,12 @@ RegisterOperators reg({
           },                                                               \
           aliasAnalysisFromSchema()),
 
-    DEFINE_TORCH_TENSOR_OP(float, double, at::scalar_to_tensor(scalar_val))
+    DEFINE_TORCH_TENSOR_OP(
+        float,
+        double,
+        at::native::scalar_tensor(
+            scalar_val,
+            at::device(at::kCPU).dtype(c10::get_default_dtype())))
```

Comment:
Is this cool for CUDA?

Reply:
yea, CUDA conversion happens after the tensor is initially constructed (as with pytorch eager)